### DISCUSSION TITLE
Untuk membuat CSS hanya bekerja pada satu file saja di React

### DISCUSSION THREAD

**Discussion by svecx** [svecx/Flow-Docs#13] [Q&A] [answered]
Dalam React, setiap komponen memiliki scope sendiri, sehingga CSS yang diterapkan hanya akan bekerja pada komponen tersebut. Berikut adalah beberapa cara untuk mengimplementasikannya

**Comment by sunshinebecomesyou** [comment, marked as answer]
Menggunakan ClassName: Anda dapat menambahkan class pada elemen HTML dalam komponen React dan menulis CSS untuk class tersebut. Ini akan memastikan bahwa styling hanya berlaku untuk elemen dalam komponen tersebut.

**Reply by svecx** [reply]
> Menggunakan ClassName: Anda dapat menambahkan class pada elemen HTML dalam komponen React dan menulis CSS untuk class tersebut. Ini akan memastikan bahwa styling hanya berlaku untuk elemen dalam komponen tersebut.

Menggunakan CSS-in-JS Library: Anda juga dapat menggunakan library CSS-in-JS seperti styled-components untuk membuat styling hanya berlaku pada komponen tertentu.
javascript
Copy code
import React from "react";
import styled from "styled-components";

const MyComponentWrapper = styled.div`
  background-color: lightgray;
  padding: 20px;
  text-align: center;
`;

function MyComponent() {
  return (
    <MyComponentWrapper>
      <p>Ini adalah contoh komponen</p>
    </MyComponentWrapper>
  );
}

export default MyComponent;
